### PR TITLE
BaSP-TG-2: Get and create admins

### PR DESCRIPTION
**Pull request by silveriac**
Get and create methods for admin users.
First two commits are wrongly numbered.

### CODE REVIEW

File: src/validations/user.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+const validateCreation = (req, res, next) => {
+  const userValidation = Joi.object({
+    firstName: Joi.string().regex(/^[a-zA-Z]+$/).min(2).max(30)
+      .required()
+      .messages({
+        'string.empty': 'first name required',
+        'string.pattern.base': 'first name should be letters only',
+        'string.min': 'first name should have a minimum length of 2 characters',
+        'string.max': 'first name should have a maximum length of 30 characters',
```
Comment (nicolasblasco) [inline src/validations/user.js:11]: The empty and required message should be the same

File: src/validations/user.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      }),
+    lastName: Joi.string().regex(/^[a-zA-Z]+$/).min(2).max(30)
+      .required()
+      .messages({
+        'string.empty': 'last name required',
+        'string.pattern.base': 'last name should be letters only',
+        'string.min': 'last name should have a minimum length of 2 characters',
+        'string.max': 'last name should have a maximum length of 30 characters',
+        'any.required': 'last name is a required field',
+      }),
```
Comment (nicolasblasco) [inline src/validations/user.js:22]: Same as above

File: src/validations/user.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    password: Joi.string().alphanum().min(8).max(50)
+      .required()
+      .messages({
+        'string.empty': 'password required',
+        'string.alphanum': 'password must be letters and numbers only',
+        'string.min': 'password should have a minimum length of 8 characters',
+        'string.max': 'password should have a maximum length of 50 characters',
+        'any.required': 'password is a required field',
+      }),
```
Comment (nicolasblasco) [inline src/validations/user.js:37]: Same as above

File: src/validations/user.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    dni: Joi.string().regex(/^\d+$/).min(7).max(11)
+      .required()
+      .messages({
+        'string.empty': 'dni required',
+        'string.pattern.base': 'dni must be numbers only',
+        'string.min': 'dni should have a minimum length of 7 characters',
+        'string.max': 'dni should have a maximum length of 11 characters',
+        'any.required': 'dni is a required field',
+      }),
```
Comment (nicolasblasco) [inline src/validations/user.js:46]: Same as above

File: src/models/Admins.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+const { Schema } = mongoose;
+
+const adminSchema = new Schema({
+  firstName: { type: String, required: true },
+  lastName: { type: String, required: true },
+  email: { type: String, required: true },
+  password: { type: String, required: true },
+  dni: { type: Number, required: true },
+  phone: { type: Number, required: true },
+  location: { type: String, required: true },
```
Comment (nicolasblasco) [inline src/models/Admins.js:12]: Phone and location shouldn't be required

File: src/controllers/admins.js [hunk 6]
```diff
@@ -1,166 +1,65 @@
...
-
-function getMaxId(adminInput) {
-  let maxId = 0;
-  adminInput.forEach((admin) => {
-    if (admin.id > maxId) {
-      maxId = admin.id;
+const getAllAdmins = async (req, res) => {
+  try {
+    const admins = await Admins.find();
+    if (!admins[0]) {
```
Comment (nicolasblasco) [inline src/controllers/admins.js:6]: The condition should be if(!admins)

File: src/controllers/admins.js [hunk 7]
```diff
@@ -1,166 +1,65 @@
...
-      });
-    } else {
-      res.status(201).json({
-        success: true,
-        msg: 'Admin created Successfully',
-        data: finalAdmin,
+};
+const getAdminById = async (req, res) => {
+  try {
+    const admin = await Admins.findById(req.params.id);
```
Comment (nicolasblasco) [inline src/controllers/admins.js:26]: We should validate that the id is a valid object id

**Review by nicolasblasco — CHANGES_REQUESTED** [review]

File: src/controllers/admins.js [hunk 8]
```diff
@@ -1,166 +1,65 @@
...
-router.get('/:id', (req, res) => {
-  const adminId = parseInt(req.params.id, 10);
-  if (!adminId) {
-    res.status(400).json({
-      success: false,
-      msg: 'The id is not valid',
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Unexpected error',
+      error,
```
Comment (nicolasblasco) [inline src/controllers/admins.js:20]: We could use the message to show all the message, like message: `Unexpected error ${error}`

File: src/controllers/admins.js [hunk 9]
```diff
@@ -1,166 +1,65 @@
...
-      msg: 'There is no Admin with this id',
+    return res.status(200).json({
+      message: 'Admin found',
+      data: admin,
+      error: false,
+    });
+  } catch (error) {
+    return res.status(500).json({
+      message: 'Unexpected error',
+      error,
```
Comment (nicolasblasco) [inline src/controllers/admins.js:41]: Same as other comment

File: src/controllers/admins.js (hunk 6 above)
Comment (silveriac) [reply src/controllers/admins.js:6]: Alternatively, I can use `if (admins.length == 0)`

**Comment by silveriac** [comment]
First two commits are  wrongly numbered

**Review by Alfondalix — APPROVED** [review]

**Pull request merged**